### DISCUSSION TITLE
[BP#62] Eviter la ré-écriture des getter / setter natifs

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#65] [Bonnes pratiques] [answered]
L'exemple me parle mais la formulation autour de getters/setters "natifs" ou de "surcharge" de getters/setters me semble pas très claire.

Dans l'exemple, le "private int" n'est pas un getter/setter à mon sens, seulement un attribut modifiable. Getter/setter sont généralement des fonctions, ça me semble confus de parler de getter/setter "natifs".

A noter qu'en terme que cette BP a aussi de bonnes raisons d'être niveau maintenabilité et sens métier.

Bon après j'ai un doute sur l'effet réel de cette BP (comme d'autres) qui relève peut-être de la micro-optimisation ?

**Comment by didierfred** [comment]
Avant d'avoir un impact mesurable, faut faire beaucoup d'appel au setter ou getter ... 

Par ailleurs si il y a des setters/getter natifs , la plupart du temps le developpeur ne va pas pour le plaisir les surcharger .

Je serais supprimer cette bonne pratique

**Comment by ddebarbieux** [comment]
Je suis aussi pour la supression. La notion de getter/setter natif est très dépendante du langage de programation, ce que nous voulons éviter dans la V4.

**Comment by ACTLEM** [comment]
Note Equipe Projet: Micro optimisation, suppression.

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/132
